app/controllers/concerns/foreman/controller/authentication.rb

```diff
@@ -6,12 +6,13 @@ def available_sso
   end
 
   def authenticate
-    return true if (User.current && Rails.env.test? && api_request?) ||
-                   session[:user] && (User.current = User.unscoped.find_by(id: session[:user]))
+    User.current = User.unscoped.find_by(id: session[:user]) if session[:user]
+    return true if (User.current && Rails.env.test? && api_request?) || User.current && !User.current.disabled?
+    return false if User.current.disabled?
```

Comment:
This is approx what I meant:
```suggestion
    return true if (User.current && Rails.env.test? && api_request?)

    User.current = User.unscoped.find_by(id: session[:user]) if session[:user]
    return !User.current.disabled? if User.current
```
And the tests are passing with this.